packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

```diff
@@ -62,9 +62,23 @@ export interface IVpcNetwork extends IConstruct {
 
   /**
    * Return the subnets appropriate for the placement strategy
+   *
+   * Prefer using {@link subnetIds} if you need to pass subnet IDs to a
+   * CloudFormation Resource.
```

Comment:
What do you mean "pass subnet IDs to a CloudFormation Resource"? It feels to me like this is all you'll ever be doing with subnets. Probably deserves a little hint at why it is preferable to use `subnetIds`, so the user feels like they understand the guidance.

Reply:
You would expect, but this is not true. What also happens to them is testing for publicness, and adding dependencies to their internetconnectivity.

A fair argument would be that those should be changed to [SOA instead of AOS](https://en.wikipedia.org/wiki/AOS_and_SOA) accessors as well, but I wanted to keep the change small.

How do you feel about those?

Reply:
Hmm yeah I think that might offer a cleaner API overall, and might allow to focus the user better on what they should do.